internal/cloud/cloud_username_manager.go

```diff
@@ -63,6 +126,15 @@ func (c *CloudUsernameManager) OnChange(ctx context.Context, st store.RStore) {
 	currentlyMakingRequest := c.currentlyMakingRequest
 	c.mu.Unlock()
 
+	// if a refresh has been induced, then do a long_get lookup, so that we get the username as soon as
+	// the user has finished the process
+	if state.TiltCloudUsernameNeedsRefresh && !currentlyMakingRequest {
+		go c.CheckUsername(ctx, st, true)
+		return
+	}
+
+	// otherwise, we're not necessarily expecting a username, so just get the current state
```

Comment:
see also: note about naming "refresh" vs. "wait for register and get username", but this comment is confusing. Maybe:
> otherwise, we don't think the user is actively in the process of registering their token; just get the current state, w/o waiting for a registration flow to finish